pkg/utils/k8sutil_test.go

```diff
@@ -1076,3 +1078,67 @@ func TestMergeDeployments(t *testing.T) {
 	}
 
 }
+
+func TestGetAnnotationsFromCRD(t *testing.T) {
+	crdWithoutAnnotationName := "crdWithoutAnnotation"
+	crdWithAnnotationName := "crdWithAnnotation"
+	expectedAnnotations := map[string]string{
+		"foo": "bar",
+	}
+	crdWithAnnotation := &extensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				"foo": "bar",
+			},
+			Name: crdWithAnnotationName,
+		},
+		Spec: extensionsv1beta1.CustomResourceDefinitionSpec{
+			Group: "foo.group.io",
+			Names: extensionsv1beta1.CustomResourceDefinitionNames{
+				Plural:   "foos",
+				Singular: "foo",
+				Kind:     "fooKind",
+				ListKind: "fooList",
+			},
+		},
+	}
+	clientset := fakeextensionsapi.NewSimpleClientset()
+	_, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crdWithAnnotation)
+	if err != nil {
+		t.Fatalf("Error while creating CRD: %v", err)
+	}
+	annotations, err := GetAnnotationsFromCRD(clientset, crdWithAnnotationName)
+	if err != nil {
+		t.Fatalf("Error while fetching CRD: %v", err)
+	}
+	for i := range expectedAnnotations {
+		if annotations[i] != expectedAnnotations[i] {
+			t.Errorf("Expecting annotation %s but received %s", expectedAnnotations[i], annotations[i])
+		}
+	}
+
+	crdWithoutAnnotation := &extensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{},
+			Name:        crdWithoutAnnotationName,
+		},
+		Spec: extensionsv1beta1.CustomResourceDefinitionSpec{
+			Group: "foo.group.io",
+			Names: extensionsv1beta1.CustomResourceDefinitionNames{
+				Plural:   "foos",
+				Singular: "foo",
+				Kind:     "fooKind",
+				ListKind: "fooList",
+			},
+		},
+	}
+	_, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crdWithoutAnnotation)
+	if err != nil {
+		t.Fatalf("Error while creating CRD: %v", err)
+	}
+	_, err = GetAnnotationsFromCRD(clientset, crdWithoutAnnotationName)
```

Comment:
I don't think we should return an error if there is no annotations, it is a valid output